lib/Doctrine/Migrations/Metadata/Storage/TableMetadataStorage.php

```diff
@@ -124,25 +125,45 @@ public function reset(): void
         );
     }
 
-    public function complete(ExecutionResult $result): void
+    /**
+     * {@inheritDoc}
+     */
+    public function complete(ExecutionResult $result, bool $dryRun = false): array
```

Comment:
Having boolean arguments is a known code smell, it often results in the method doing too many things. Here we can see how it makes the complexity skyrocket. I think you should create a new method.

Reply:
totally agree. new method MetadataStorage::getSql() added